### ISSUE TITLE
Redirect stderr to stdout?

### ISSUE THREAD

**Issue by ianthehenry** [andrewchambers/janet-sh#17]
I would expect this to work, but it doesn't:

```
repl:1:> (use sh)
...
repl:2:> ($< sh -c `echo hi >&2` > [stderr stdout])
hi
""
```

(I know I can capture stderr by redirecting it to a buffer -- the `$<` is just to demonstrate that it didn't actually do the thing.)

Is there a way to redirect stderr to stdout?

**Comment by andrewchambers** [comment]
I thought this should work, seems like a bug potentially if it doesn't.

**Comment by andrewchambers** [comment]
I would also try swapping the stderr and stdout - I think its slightly counter intuitive - its essentially equivalent to a call to dup2 iirc.

**Comment by ianthehenry** [comment]
Yeah, swapping doesn't seem to make a difference:

```
repl:2:> ($< sh -c `echo hi >&2` > [stdout stderr])
hi
""
repl:3:> ($< sh -c `echo hi >&2` > [stderr stdout])
hi
""
```

**Comment by ianthehenry** [comment]
So the redirect was working correctly, and using `sh/$` for example gave the correct behavior. The problem was that the implicit `dup2` done by `$<` happens after this, so for a brief time stderr and stdout do refer to the same file, but the second call to `dup2` causes stdout to point elsewhere. If I'm understanding this right. This appears to be easily fixed by #19.
